source/particle/world.cc

```diff
@@ -568,116 +560,113 @@ namespace aspect
       // because it only knows the subdomain_id of ghost cells, but not
       // of artificial cells.
 
-      particle_timer->enter_section("Sort particles");
-
       // There are three reasons why a particle is not in its old cell:
       // It moved to another cell, to another domain or it left the mesh.
       // Sort the particles accordingly and deal with them
       std::multimap<types::LevelInd, Particle<dim> >     moved_particles_cell;
       std::multimap<types::subdomain_id, Particle<dim> > moved_particles_domain;
       std::multimap<types::LevelInd, Particle<dim> >     lost_particles;
 
-      // Find the cells that the particles moved to.
-      // Note that the iterator in the following loop is increased in a
-      // very particular way, because it is changed, if elements
-      // get erased. A change can result in invalid memory access.
-      typename std::multimap<types::LevelInd, Particle<dim> >::iterator   it;
-      for (it=particles.begin(); it!=particles.end();)
-        {
-          // The cell the particle is in
-          typename parallel::distributed::Triangulation<dim>::active_cell_iterator current_cell;
-          bool found_cell = false;
-
-          // If we know the particle's old cell, check if it is still inside
-          // or in one of its neighbors
-          if (it->first != std::make_pair(-1,-1))
-            {
-              current_cell = typename parallel::distributed::Triangulation<dim>::active_cell_iterator (&(this->get_triangulation()), it->first.first, it->first.second);
+      TimerOutput::Scope timer_section(*particle_timer,"Particles: Sort");
+      {
```

Comment:
I suspect that you wanted the timer variable inside the block you define with the curly braces.